markdown/frameworks/framework_einstieg.md

```diff
@@ -0,0 +1,146 @@
+---
+type: lecture-cg
+title: "Frameworks: Einstieg"
+menuTitle: "Frameworks"
+author: "André Matutat (FH Bielefeld)"
+weight: 1
+readings:
+  - key: ""
+tldr: |
+  Frameworks sind ein Softwaregerüst, um wiederkehrende Softwarestrukturen schnell und einfach um die eigene Funktionalität zu erweitern.
+  Im Gegensatz zu Libraries stellen sie keine einzelnen Funktionen bereits, die im eigene Code aufgerufen werden, sondern geben Strukturen vor, in denen der eigenen Code eingearbeitet werden muss.
+outcomes:
+  - k1: "Begriff Framework kennen gelernt"
+  - k2: "Unterschied zwischen Frameworks und Libraries verstehen"
+quizzes:
+  - link: "XYZ"
+    name: "Quiz XXX (ILIAS)"
+youtube:
+  - link: ""
+    name: "VL "
+  - link: ""
+    name: "Demo "
+  - link: ""
+    name: "Demo "
+fhmedia:
+  - link: ""
+    name: "VL "
+sketch: true
+---
+
+
+## Motivation
+
+Neben der eigentlichen Funktionalität werden an Softwareprodukte viele weitere Randbindungen gestellt.
+Dies können Sicherheitsbindungen sein oder Anforderungen an die Art und Weise der Bereitstellung des Produktes, zum Beispiel als Webservice.
+Viele dieser Anforderungen ähneln sich von Produkt zu Produkt und müssen dennoch immer wieder neu programmiert werden.
+
+Für einen Webservice muss eine HTML-Schnittstelle bereitgestellt werden, die Anfragen annimmt, bearbeitet und die Antworten wieder zurückliefert.
+
+## Was sind Frameworks?
+
+> Ein Framework (englisch für Rahmenstruktur) ist ein Programmiergerüst, das in der Softwaretechnik, insbesondere im Rahmen der objektorientierten Softwareentwicklung sowie bei komponentenbasierten Entwicklungsansätzen, verwendet wird. Im allgemeineren Sinne bezeichnet man mit Framework auch einen Ordnungsrahmen. [https://de.wikipedia.org/wiki/Framework]
+
+Frameworks liefern die Rahmenstruktur und Architektur, um Programme für die verschiedenen Anwendungszwecke einfacher und effizient zu entwickeln. Sie legen dabei fest, WIE programmiert werden soll.
+
+Frameworks stellen Code bereit, der an entsprechenden Stellen mit eigenen Klassen und Funktionen erweitert werden muss, um die eigentliche Funktion der Anwendung zu implementieren, sind dabei aber selbst keine eigenen Programme (vgl. PM-Dungeon).
+
+Frameworks sind die Lückentexte der Programmierung. <!-- //Ich finde die Analogie eigentlich nicht schlecht, aber die muss besser eingebunden werden-->
+
+
+## Unterschied zu Libraries
```

Comment:
Das ist super! Kannst Du das graphisch darstellen? (Den Text drin lassen und über `::: notes ... :::` nur im Skript erscheinen lassen.)

Ein "UML"-Bild mit ner linken und ner rechten Seite, Format 16:10 (ca.). Links die Lib, rechts das Framework. Wegen mir auch mit Draw.io ...

Reply:
Dachte eigentlich ich hab das mit "frameworksVSlib" schon graphisch dargestellt. Oder willst du was technischeres (Fragte er ohne einen richtigen Plan zu haben, was das sein könnte)?

Reply:
ah, ich hab das bild gesehen, aber nicht in einen zusammenhang mit den slides gebracht (im diff-view sieht man das bild ja nicht gerendert)

bin noch nicht sicher, ob das bild das so klar aussagt. habe aber auch keine richtige idee, wie man das klarer machen könnte. hatte so ein bild im kopf: block framework/lib, block eigener code, pfeile der nutzungsrichtung (die dann umgedreht sind). aber so richtig besser wird das alleine auch nicht, höchstens im zusammenhang.

Reply:
Persönlich find ich das Bild recht passend. Vor allem greift es den Lückentext-Gedanken von weiter oben wieder auf.
Aber das ist halt auch nur meine Art zu denken.

Reply:
wenn man drüber nachdenkt, passt das auch. ich hatte halt was mit pfeilen und abhängigkeiten und so im kopf, deshalb bin ich da erst drüber gestolpert. 

wir lassen es einfach bei dem bild von dir.